### ISSUE TITLE
if a template is a struct template then the comment will be broken.

### ISSUE THREAD

**Issue by peaceshi** [jeff-hykin/cpp-textmate-grammar#82]
maybe this is also the txetmate bug.

![image](https://user-images.githubusercontent.com/10380828/55923302-7953c300-5c37-11e9-88f8-feb11c45a67a.png)
https://github.com/lhmouse/asteria/blob/master/asteria/src/rocket/cow_hashmap.hpp#L200

**Comment by jeff-hykin** [comment]
This might be a duplicate of #35, take a look at it if you want more info.

**Comment by jeff-hykin** [comment]
@peaceshi You have some mighty syntax-complicated code haha. It's actually helpful though, thanks for posting all of these.

**Comment by peaceshi** [comment]
> @peaceshi You have some mighty syntax-complicated code haha. It's actually helpful though, thanks for posting all of these.

haha , it is my friend's project. He is the author of `GCC with the MCF thread model`.
https://mingw-w64.org/doku.php/download

**Comment by jeff-hykin** [comment]
Wow 😮 thats some serious stuff.

On second note, I think this is actually not related to #35 
```
struct copy_storage_helper<pointerT, hashT, allocatorT, false>     // copyableT
      {
        [[noreturn]] void operator()(pointerT /*ptr*/, const hashT& /*hf*/, pointerT /*ptr_old*/, size_t /*off*/, size_t /*cnt*/) const
          {
            // `allocatorT::value_type` is not copy-constructible.
            // Throw an exception unconditionally, even when there is nothing to copy.
            noadl::sprintf_and_throw<domain_error>("cow_hashmap: `%s` is not copy-constructible.",
                                                   typeid(typename allocatorT::value_type).name());
          }
      };
```
Even without the template the highlighting screws up.

**Comment by jeff-hykin** [comment]
This should be fixed now `v1.7.5` both the template and the comment (which were actually separate issues).

There is a known bug where including nested `<>`'s will screw things up, but its nearly impossible to fix.
